### PR TITLE
Add artifact handler

### PR DESCRIPTION
**Pull request by owlishDeveloper**
This is a concept  of my solution for https://github.com/mozilla-mobile/fenix/issues/6760
I am looking for some feedback on the concept (for example, if this particular way is not great, I'd like an explanation _why_, preferably without a suggestion how to do it instead; if I am missing a crucial piece - please let me know; etc.)

### CODE REVIEW

**Comment by owlishDeveloper** [comment]
> Most artifacts are created when a task is completed, and in this implementation parsing of the live log does not complete until the task is finished

When is the artifact-created message is sent? The process of creating an artifact takes some time, so I was assuming the message is sent at the end of the process, and not at the beginning?

**Comment by djmitche** [comment]
> When is the artifact-created message is sent? The process of creating an artifact takes some time, so I was assuming the message is sent at the end of the process, and not at the beginning?

It's sent when the artifact is readable from the API.

**Comment by owlishDeveloper** [comment]
> Most artifacts are created when a task is completed, and in this implementation parsing of the live log does not complete until the task is finished

When is the artifact-created message is sent? The creation of the artifact takes some time, and I assumed the message was sent at the end of the process as opposed to beginning. Is that not accurate?

File: services/github/src/handlers.js [hunk 1]
```diff
@@ -863,3 +884,98 @@ async function taskDefinedHandler(message) {
...
+    .find(a => a.storageType === 'reference')
+    .name;
+  let latestLog = await this.queueClient.getLatestArtifact(taskId, artifactName);
+
+  // 3. parse the logs
+  const parseExecTests = 'EXECUTE TEST(S)';
+  const parseDevice = 'device';
+  const parseFlank = 'flank';
+  const parseMatrixRes = 'MatrixResultsReport';
+  const parseTaskFinished = '=== Task Finished ===';
```
[the remark this answers is not in the document]
Comment (owlishDeveloper) [reply services/github/src/handlers.js]: Yes, I was warning about this at the All Hands. The first iteration is supposed to be project-specific, and then could be iterated upon to make it more generic

**Comment by owlishDeveloper** [comment]
> > When is the artifact-created message is sent? The process of creating an artifact takes some time, so I was assuming the message is sent at the end of the process, and not at the beginning?
> 
> It's sent when the artifact is readable from the API.

And at which point in the creation process does it become readable from the API?

**Comment by djmitche** [comment]
> And at which point in the creation process does it become readable from the API?

For Reference artifacts, it's once the artifact is in the backend (Azure, at the moment) storage.

I think there's a race condition for S3 artifacts, but I'll leave that for you to investigate.

File: services/github/src/handlers.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (owlishDeveloper) [reply services/github/src/handlers.js]: (We discussed this elsewhere and I'll make it more more generic)

**Comment by owlishDeveloper** [comment]
I'm closing this as I am back to the drawing board. Thank you @djmitche !

**Pull request closed without merging**